web/source/keyboards/kmwkeyboards.ts

```diff
@@ -1209,24 +1220,37 @@ namespace com.keyman.keyboards {
             }
           }
           if (!languageFound) {
-            // TODO: Construct response array of errors (failed-query keyboards)
+            // Construct response array of errors (failed-query keyboards)
             // that will be merged with stubs (successfully-queried keyboards)
-            console.error(this.alertLanguageUnavailable(lgName));
+            let stub: ErrorStub = {language: {name: lgName}, error: new Error(this.alertLanguageUnavailable(lgName))};
+            errorStub.push(stub);
           }
         }
 
         if(cmd == '') {
-          return Promise.reject(new Error(this.alertLanguageUnavailable(languages[0])));
+          // No command so return errors
+          return Promise.reject(errorStub);
         } else {
-          return this.keymanCloudRequest('&keyboardid='+cmd, false);
+          // Merge this with errorStub
+          let result = await this.keymanCloudRequest('&keyboardid='+cmd, false);
+          if (result instanceof Error) {
```

Comment:
How would this receive an `instanceof Error`?  TS tells me

`let result: (KeyboardStub | ErrorStub)[]`

when I inspect the code in my editor.

Either this should never occur... or the typing's gone wrong somewhere.

Reply:
If you're trying to 'catch' a `reject` error from the Promise, that's not quite the way to do it.

https://itnext.io/error-handling-with-async-await-in-js-26c3f20bc06a

It seems that a wrapping try-catch is in order when async is used.